frontend/src/context/cart/CartProvider.tsx

```diff
@@ -0,0 +1,47 @@
+import { createContext, PropsWithChildren, useCallback, useMemo, useState } from 'react';
+
+import { TItem } from '@/types/cart';
+
+type InitValue = {
+  items: TItem[];
+  addItem: (product: TItem) => void;
+  deleteItem: (id: number) => void;
+};
+
+const initValue: InitValue = {
+  items: [],
+  addItem: () => null,
+  deleteItem: () => null,
+};
+
+export const CartContext = createContext(initValue);
+
+export default function CartProvider({ children }: PropsWithChildren) {
+  const [items, setItems] = useState<TItem[]>([]);
+
+  const deleteItem = useCallback(
+    (id: number) => {
+      const filteredItems = items.filter((item) => item.id !== id);
+      setItems(filteredItems);
```

Comment:
callback으로 만들면 deps 배열에 items를 의존하지 않아도 될 것 같군요!

```suggestion
      setItems(prev => prev.filter((item) => item.id !== id));
```